standard/structs.md

```diff
@@ -476,3 +501,71 @@ Static constructors for structs follow most of the same rules as for classes. Th
 Automatically implemented properties ([§15.7.4](classes.md#1574-automatically-implemented-properties)) use hidden backing fields, which are only accessible to the property accessors.
 
 > *Note*: This access restriction means that constructors in structs containing automatically implemented properties often need an explicit constructor initializer where they would not otherwise need one, to satisfy the requirement of all fields being definitely assigned before any function member is invoked or the constructor returns. *end note*
+
+### §safe-scope-rules Safe scope constraint for `ref struct` types
+
+#### safe-scope-rules-general General
+
+At compile-time, each expression whose type is a ref struct is associated with a scope where that instance is safe, its ***safe-scope***. The safe-scope is a scope, enclosing an expression, which it is safe for the value to escape to. If that scope is the entire function member, we say that the value is ***safe-to-return*** from the function member.
+
+The safe-scope records which scope a ref struct may be copied into. Given an assignment from an expression `E1` with a safe-scope `S1`, to an expression `E2` with safe-scope `S2`, it is an error if `S2` is a wider scope than `S1`.
+
+There are three different safe-scope values, the same as the ref-safe-scope values defined for reference variables (§ref-span-safety-escape-scopes): block, function-member, and caller-scope. An expression `e1` of a ref struct type is constrained by its safe-scope as follows:
+
+- For a return statement `return e1`, the safe-scope of `e1` must be calling-method.
+- For an assignment `e1 = e2` the safe-scope of `e2` must be at least as wide a scope as the safe-scope of `e1`.
+
+For a method invocation if there is a `ref` or `out` argument of a `ref struct` type (including the receiver unless the type is `readonly`), with safe-scope `S1`, then no argument (including the receiver) may have a narrower safe-scope than `S1`.
+
+Any expression whose compile-time type is not a ref struct has a safe-scope of calling-method.
+
+A `default` expression, for any type, has safe-scope of calling-method.
+
+#### safe-scope-rules-parameter Parameter safe scope
+
+A formal parameter of a ref struct type, including the `this` parameter of an instance method, has a safe-scope of calling-method.
+
+#### safe-scope-rules-local Local variable safe scope
+
+A local variable of a ref struct type has a safe-scope as follows:
+
+- If the variable is an iteration variable of a `foreach` loop, then the variable's safe-scope is the same as the safe-scope of the `foreach` loop's expression.
+- A local of `ref struct` type which is uninitialized at the point of declaration has a safe-scope of calling method.
+- Otherwise the variable's declaration requires an initializer. The variable's safe-scope is the same as the safe-scope of its initializer.
```

Comment:
There is a bit of repetition here and unusual phrasing, maybe try:

```suggestion
A local variable of a ref struct type has a safe-scope as follows:

- If the variable is an iteration variable of a `foreach` loop, then the variable's safe-scope is the same as the safe-scope of the `foreach` loop's expression.
- Otherwise if the variable's declaration has an initializer then the variable's safe-scope is the same as the safe-scope of that initializer.

```

Note: I omitted that the final option means the variable is not initialised, if that is desired the final bullet can be:

```
- Otherwise the variable is uninitialized at the point of declaration and has a safe-scope of the calling method.
```